<h1 style='background:#483D8B; border:2; border-radius: 10px; font-size:250%; font-weight: bold; color:white'><center>InfoCepts Data Engineering Hiring Hackathon<br> Author : Abhineet Kumar
    </center></h1>

### **<span style="color:#483D8B;">import libraries</span>**

In [1]:
def warn(*args,**kwargs):
    pass
import warnings
warnings.warn = warn
import numpy as np
import pandas as pd
import zipfile

### **<span style="color:#483D8B;">Read Data</span>**

### unzipping folder </br>
with zipfile.ZipFile("../Hackathon/InfoCept Participants Data.zip","r") as zip_ref: </br>
    zip_ref.extractall("../Hackathon/") 

In [2]:
df=pd.read_xml("../Hackathon/InfoCept Participants Data/infocept_xml_data_.xml")
df.head()

index                          user_session  total_clicks  total_items  \
0      0  8b69208e-7c64-422d-b670-5b9d82d25606             7            2   
1      1  8068596c-a2b8-46ca-8791-e10d70520a1f             7            2   
2      2  4d153217-e06d-4581-8924-702c7f757610             2           22   
3      3  e65a6869-fb45-4f34-8482-c80278676d0f             2            2   
4      4  d8206742-06d7-42a2-8163-d70f9dd6cb18             5            1   

   total_cats  max_dwell  mean_dwell  total_duration  click_rate  day_of_week  \
0           1   0.511508    5.995878      166.711175    0.072524            1   
1           1   7.626576    3.529165        0.434185    0.204248            0   
2           1   3.095637  180.977516        0.935320    0.045982            4   
3           1   1.089128    4.338280        0.417055    0.028476            0   
4           1   0.329814    3.180891        2.485720    0.094812            5   

   ...      cat_most_viewed  prod_most_viewed_n_times  prod_most_viewed  \
0  ...  2053013560982667520                       1.0           1030297   
1  ...  2053013602328876544                       1.0          36083228   
2  ...  2175094838038844672                       2.0           1002263   
3  ...  2053013568140556288                       3.0           1003257   
4  ...  2053013558410492160                       1.0          13728056   

   cat_views_freqs  prod_views_freqs  cat_buys_freqs  prod_buys_freqs  \
0         0.267802          0.000945    2.337402e-01         0.000690   
1         0.140925          0.000946    4.550292e-01         0.000690   
2         0.000294          0.000945    9.102017e-08         0.000690   
3         0.150107          0.000945    4.550279e-01         0.000739   
4         0.267784          0.000945    4.550284e-01         0.000690   

   is_purchase                         start_time_ts  \
0            0     October 07, 2019 Monday, 00:08:28   
1            0    October 01, 2019 Tuesday, 15:08:40   
2            0     October 14, 2019 Monday, 21:17:24   
3            0   October 05, 2019 Saturday, 21:54:13   
4            0  October 09, 2019 Wednesday, 13:33:22   

                            end_time_ts  
0   October 26, 2019 Saturday, 16:08:29  
1    October 15, 2019 Tuesday, 20:03:18  
2  October 23, 2019 Wednesday, 21:56:30  
3   October 17, 2019 Thursday, 13:54:29  
4    October 29, 2019 Tuesday, 11:51:04  

[5 rows x 24 columns]

In [3]:
df.columns

Index(['index', 'user_session', 'total_clicks', 'total_items', 'total_cats',
       'max_dwell', 'mean_dwell', 'total_duration', 'click_rate',
       'day_of_week', 'is_weekend', 'is_special_day', 'time_of_day',
       'cat_most_viewed_n_times', 'cat_most_viewed',
       'prod_most_viewed_n_times', 'prod_most_viewed', 'cat_views_freqs',
       'prod_views_freqs', 'cat_buys_freqs', 'prod_buys_freqs', 'is_purchase',
       'start_time_ts', 'end_time_ts'],
      dtype='object')

In [4]:
df['start_time_ts']=pd.to_datetime(df['start_time_ts'])
df['end_time_ts']=pd.to_datetime(df['end_time_ts'])
df

index                          user_session  total_clicks  \
0            0  8b69208e-7c64-422d-b670-5b9d82d25606             7   
1            1  8068596c-a2b8-46ca-8791-e10d70520a1f             7   
2            2  4d153217-e06d-4581-8924-702c7f757610             2   
3            3  e65a6869-fb45-4f34-8482-c80278676d0f             2   
4            4  d8206742-06d7-42a2-8163-d70f9dd6cb18             5   
...        ...                                   ...           ...   
499995  499995  fee0375f-7432-4912-87dd-355e58ff748e             7   
499996  499996  3dc39db9-b779-4475-a3e7-5e8c63c59e93             2   
499997  499997  d619f46b-cd01-459f-bcac-950aaea63627             6   
499998  499998  efb4fdc8-b413-4c64-9ac3-a0033b091008             4   
499999  499999  9ae3ff1c-7608-49f7-a227-e948365715f9             3   

        total_items  total_cats  max_dwell  mean_dwell  total_duration  \
0                 2           1   0.511508    5.995878      166.711175   
1                 2           1   7.626576    3.529165        0.434185   
2                22           1   3.095637  180.977516        0.935320   
3                 2           1   1.089128    4.338280        0.417055   
4                 1           1   0.329814    3.180891        2.485720   
...             ...         ...        ...         ...             ...   
499995            2           1   5.807861    0.135879        0.222842   
499996            2           1   0.679914    2.848078        6.560991   
499997            3           1   0.840349    1.517229        0.023602   
499998            2           1   3.396826    3.365726        0.020893   
499999            2           1   6.252938    1.960957        0.024305   

        click_rate  day_of_week  ...      cat_most_viewed  \
0         0.072524            1  ...  2053013560982667520   
1         0.204248            0  ...  2053013602328876544   
2         0.045982            4  ...  2175094838038844672   
3         0.028476            0  ...  2053013568140556288   
4         0.094812            5  ...  2053013558410492160   
...            ...          ...  ...                  ...   
499995    0.116420            5  ...  2053013565659966976   
499996    0.072760            5  ...  2053013560323437056   
499997    0.055101            5  ...  2053013561594823168   
499998    0.089165            5  ...  2053013558504300032   
499999    0.078291            5  ...  2053013577775031552   

        prod_most_viewed_n_times  prod_most_viewed  cat_views_freqs  \
0                            1.0           1030297         0.267802   
1                            1.0          36083228         0.140925   
2                            2.0           1002263         0.000294   
3                            3.0           1003257         0.150107   
4                            1.0          13728056         0.267784   
...                          ...               ...              ...   
499995                       3.0          30023271         0.000002   
499996                       1.0          12704844         0.000002   
499997                       3.0           1014692         0.267806   
499998                       1.0          19506952         0.000009   
499999                       1.0           1021591         0.267804   

        prod_views_freqs  cat_buys_freqs  prod_buys_freqs  is_purchase  \
0               0.000945    2.337402e-01         0.000690            0   
1               0.000946    4.550292e-01         0.000690            0   
2               0.000945    9.102017e-08         0.000690            0   
3               0.000945    4.550279e-01         0.000739            0   
4               0.000945    4.550284e-01         0.000690            0   
...                  ...             ...              ...          ...   
499995          0.000945    0.000000e+00         0.000689            1   
499996          0.000945    1.294499e-04         0.000691            1   
499997          0.000962   

***1. Process the "start_time_ts" column to create "time_spec_points" column. Allot exact points equaling the time of the day the user is surfing, but allot extra 5 points if they are surfing after 4 o clock in the afternoon. For example: If the user starts surfing at 9 o clock in the morning allot 9 points, but user surfing after 4 o clock in the afternoon will receive 21 points.***

In [5]:
df['hour']=df['start_time_ts'].dt.hour
df.head(2)

index                          user_session  total_clicks  total_items  \
0      0  8b69208e-7c64-422d-b670-5b9d82d25606             7            2   
1      1  8068596c-a2b8-46ca-8791-e10d70520a1f             7            2   

   total_cats  max_dwell  mean_dwell  total_duration  click_rate  day_of_week  \
0           1   0.511508    5.995878      166.711175    0.072524            1   
1           1   7.626576    3.529165        0.434185    0.204248            0   

   ...  prod_most_viewed_n_times  prod_most_viewed  cat_views_freqs  \
0  ...                       1.0           1030297         0.267802   
1  ...                       1.0          36083228         0.140925   

   prod_views_freqs  cat_buys_freqs  prod_buys_freqs  is_purchase  \
0          0.000945        0.233740          0.00069            0   
1          0.000946        0.455029          0.00069            0   

        start_time_ts         end_time_ts  hour  
0 2019-10-07 00:08:28 2019-10-26 16:08:29     0  
1 2019-10-01 15:08:40 2019-10-15 20:03:18    15  

[2 rows x 25 columns]

In [6]:
def Time_spec(t):
    if (t<16):
        return t
    elif t>=16:
        return t+5

In [7]:
df['time_spec_points']=df['hour'].apply(lambda x:Time_spec(x))
df.head()

index                          user_session  total_clicks  total_items  \
0      0  8b69208e-7c64-422d-b670-5b9d82d25606             7            2   
1      1  8068596c-a2b8-46ca-8791-e10d70520a1f             7            2   
2      2  4d153217-e06d-4581-8924-702c7f757610             2           22   
3      3  e65a6869-fb45-4f34-8482-c80278676d0f             2            2   
4      4  d8206742-06d7-42a2-8163-d70f9dd6cb18             5            1   

   total_cats  max_dwell  mean_dwell  total_duration  click_rate  day_of_week  \
0           1   0.511508    5.995878      166.711175    0.072524            1   
1           1   7.626576    3.529165        0.434185    0.204248            0   
2           1   3.095637  180.977516        0.935320    0.045982            4   
3           1   1.089128    4.338280        0.417055    0.028476            0   
4           1   0.329814    3.180891        2.485720    0.094812            5   

   ...  prod_most_viewed  cat_views_freqs  prod_views_freqs  cat_buys_freqs  \
0  ...           1030297         0.267802          0.000945    2.337402e-01   
1  ...          36083228         0.140925          0.000946    4.550292e-01   
2  ...           1002263         0.000294          0.000945    9.102017e-08   
3  ...           1003257         0.150107          0.000945    4.550279e-01   
4  ...          13728056         0.267784          0.000945    4.550284e-01   

   prod_buys_freqs  is_purchase       start_time_ts         end_time_ts  hour  \
0         0.000690            0 2019-10-07 00:08:28 2019-10-26 16:08:29     0   
1         0.000690            0 2019-10-01 15:08:40 2019-10-15 20:03:18    15   
2         0.000690            0 2019-10-14 21:17:24 2019-10-23 21:56:30    21   
3         0.000739            0 2019-10-05 21:54:13 2019-10-17 13:54:29    21   
4         0.000690            0 2019-10-09 13:33:22 2019-10-29 11:51:04    13   

   time_spec_points  
0                 0  
1                15  
2                26  
3                26  
4                13  

[5 rows x 26 columns]

***2. Reduce the "total_duration" round to 2 decimal places: example (166.711175 -> 166.71) (0.935320 -> 0.94)***

In [8]:
df['total_duration']=df['total_duration'].apply(lambda x:float(np.round(x, 2)))
df.head()

index                          user_session  total_clicks  total_items  \
0      0  8b69208e-7c64-422d-b670-5b9d82d25606             7            2   
1      1  8068596c-a2b8-46ca-8791-e10d70520a1f             7            2   
2      2  4d153217-e06d-4581-8924-702c7f757610             2           22   
3      3  e65a6869-fb45-4f34-8482-c80278676d0f             2            2   
4      4  d8206742-06d7-42a2-8163-d70f9dd6cb18             5            1   

   total_cats  max_dwell  mean_dwell  total_duration  click_rate  day_of_week  \
0           1   0.511508    5.995878          166.71    0.072524            1   
1           1   7.626576    3.529165            0.43    0.204248            0   
2           1   3.095637  180.977516            0.94    0.045982            4   
3           1   1.089128    4.338280            0.42    0.028476            0   
4           1   0.329814    3.180891            2.49    0.094812            5   

   ...  prod_most_viewed  cat_views_freqs  prod_views_freqs  cat_buys_freqs  \
0  ...           1030297         0.267802          0.000945    2.337402e-01   
1  ...          36083228         0.140925          0.000946    4.550292e-01   
2  ...           1002263         0.000294          0.000945    9.102017e-08   
3  ...           1003257         0.150107          0.000945    4.550279e-01   
4  ...          13728056         0.267784          0.000945    4.550284e-01   

   prod_buys_freqs  is_purchase       start_time_ts         end_time_ts  hour  \
0         0.000690            0 2019-10-07 00:08:28 2019-10-26 16:08:29     0   
1         0.000690            0 2019-10-01 15:08:40 2019-10-15 20:03:18    15   
2         0.000690            0 2019-10-14 21:17:24 2019-10-23 21:56:30    21   
3         0.000739            0 2019-10-05 21:54:13 2019-10-17 13:54:29    21   
4         0.000690            0 2019-10-09 13:33:22 2019-10-29 11:51:04    13   

   time_spec_points  
0                 0  
1                15  
2                26  
3                26  
4                13  

[5 rows x 26 columns]

***3.  Add the values of columns  "total_clicks", "total_items", "total_cats" to create a new column: "total_inventory". If the value "total_inventory" is greater than 10, then make sure the value is increased by 100%.***

In [9]:
def get_total_inventory(clicks,items,cats):
    total=clicks+items+cats
    if total>10:
        return total*2
    else:
        return total

In [10]:
df['total_inventory'] = df[['total_clicks','total_items','total_cats']].apply(lambda x: get_total_inventory(*x), axis=1)
df.head()

index                          user_session  total_clicks  total_items  \
0      0  8b69208e-7c64-422d-b670-5b9d82d25606             7            2   
1      1  8068596c-a2b8-46ca-8791-e10d70520a1f             7            2   
2      2  4d153217-e06d-4581-8924-702c7f757610             2           22   
3      3  e65a6869-fb45-4f34-8482-c80278676d0f             2            2   
4      4  d8206742-06d7-42a2-8163-d70f9dd6cb18             5            1   

   total_cats  max_dwell  mean_dwell  total_duration  click_rate  day_of_week  \
0           1   0.511508    5.995878          166.71    0.072524            1   
1           1   7.626576    3.529165            0.43    0.204248            0   
2           1   3.095637  180.977516            0.94    0.045982            4   
3           1   1.089128    4.338280            0.42    0.028476            0   
4           1   0.329814    3.180891            2.49    0.094812            5   

   ...  cat_views_freqs  prod_views_freqs  cat_buys_freqs  prod_buys_freqs  \
0  ...         0.267802          0.000945    2.337402e-01         0.000690   
1  ...         0.140925          0.000946    4.550292e-01         0.000690   
2  ...         0.000294          0.000945    9.102017e-08         0.000690   
3  ...         0.150107          0.000945    4.550279e-01         0.000739   
4  ...         0.267784          0.000945    4.550284e-01         0.000690   

   is_purchase       start_time_ts         end_time_ts  hour  \
0            0 2019-10-07 00:08:28 2019-10-26 16:08:29     0   
1            0 2019-10-01 15:08:40 2019-10-15 20:03:18    15   
2            0 2019-10-14 21:17:24 2019-10-23 21:56:30    21   
3            0 2019-10-05 21:54:13 2019-10-17 13:54:29    21   
4            0 2019-10-09 13:33:22 2019-10-29 11:51:04    13   

   time_spec_points  total_inventory  
0                 0               10  
1                15               10  
2                26               50  
3                26                5  
4                13                7  

[5 rows x 27 columns]

***4. Create a new column "give_big_discount". The value of '"give_big_discount" should be 1 if day_of_week is 0 and is_special_day is 1, otherwise 0.***

In [11]:
def get_big_discount(DayOfWeek,Special_day):
    if (DayOfWeek==0) & (Special_day==1):
        return 1
    else:
        return 0
    
df['give_big_discount'] = df[['day_of_week','is_special_day']].apply(lambda x: get_big_discount(*x), axis=1)
df.head()

index                          user_session  total_clicks  total_items  \
0      0  8b69208e-7c64-422d-b670-5b9d82d25606             7            2   
1      1  8068596c-a2b8-46ca-8791-e10d70520a1f             7            2   
2      2  4d153217-e06d-4581-8924-702c7f757610             2           22   
3      3  e65a6869-fb45-4f34-8482-c80278676d0f             2            2   
4      4  d8206742-06d7-42a2-8163-d70f9dd6cb18             5            1   

   total_cats  max_dwell  mean_dwell  total_duration  click_rate  day_of_week  \
0           1   0.511508    5.995878          166.71    0.072524            1   
1           1   7.626576    3.529165            0.43    0.204248            0   
2           1   3.095637  180.977516            0.94    0.045982            4   
3           1   1.089128    4.338280            0.42    0.028476            0   
4           1   0.329814    3.180891            2.49    0.094812            5   

   ...  prod_views_freqs  cat_buys_freqs  prod_buys_freqs  is_purchase  \
0  ...          0.000945    2.337402e-01         0.000690            0   
1  ...          0.000946    4.550292e-01         0.000690            0   
2  ...          0.000945    9.102017e-08         0.000690            0   
3  ...          0.000945    4.550279e-01         0.000739            0   
4  ...          0.000945    4.550284e-01         0.000690            0   

        start_time_ts         end_time_ts  hour  time_spec_points  \
0 2019-10-07 00:08:28 2019-10-26 16:08:29     0                 0   
1 2019-10-01 15:08:40 2019-10-15 20:03:18    15                15   
2 2019-10-14 21:17:24 2019-10-23 21:56:30    21                26   
3 2019-10-05 21:54:13 2019-10-17 13:54:29    21                26   
4 2019-10-09 13:33:22 2019-10-29 11:51:04    13                13   

   total_inventory  give_big_discount  
0               10                  0  
1               10                  0  
2               50                  0  
3                5                  0  
4                7                  0  

[5 rows x 28 columns]

***5. Create a new column prod_views_buys_ratio which should be a ratio of prod_views_freqs and prod_buys_freqs***

In [12]:
df['prod_views_buys_ratio'] = df.prod_views_freqs.div(df.prod_buys_freqs)
df.head()

index                          user_session  total_clicks  total_items  \
0      0  8b69208e-7c64-422d-b670-5b9d82d25606             7            2   
1      1  8068596c-a2b8-46ca-8791-e10d70520a1f             7            2   
2      2  4d153217-e06d-4581-8924-702c7f757610             2           22   
3      3  e65a6869-fb45-4f34-8482-c80278676d0f             2            2   
4      4  d8206742-06d7-42a2-8163-d70f9dd6cb18             5            1   

   total_cats  max_dwell  mean_dwell  total_duration  click_rate  day_of_week  \
0           1   0.511508    5.995878          166.71    0.072524            1   
1           1   7.626576    3.529165            0.43    0.204248            0   
2           1   3.095637  180.977516            0.94    0.045982            4   
3           1   1.089128    4.338280            0.42    0.028476            0   
4           1   0.329814    3.180891            2.49    0.094812            5   

   ...  cat_buys_freqs  prod_buys_freqs  is_purchase       start_time_ts  \
0  ...    2.337402e-01         0.000690            0 2019-10-07 00:08:28   
1  ...    4.550292e-01         0.000690            0 2019-10-01 15:08:40   
2  ...    9.102017e-08         0.000690            0 2019-10-14 21:17:24   
3  ...    4.550279e-01         0.000739            0 2019-10-05 21:54:13   
4  ...    4.550284e-01         0.000690            0 2019-10-09 13:33:22   

          end_time_ts  hour  time_spec_points  total_inventory  \
0 2019-10-26 16:08:29     0                 0               10   
1 2019-10-15 20:03:18    15                15               10   
2 2019-10-23 21:56:30    21                26               50   
3 2019-10-17 13:54:29    21                26                5   
4 2019-10-29 11:51:04    13                13                7   

   give_big_discount  prod_views_buys_ratio  
0                  0               1.369726  
1                  0               1.371440  
2                  0               1.369958  
3                  0               1.279051  
4                  0               1.370596  

[5 rows x 29 columns]

***6. Create a new column create loyalty_points which should be calculated based on the following conditions: if a user is spending more than 3 seconds on a Sunday -- and loyalty points score will be 10 multiplied seconds over 3.***

In [13]:
df['day_name']=df['start_time_ts'].dt.day_name()
df.head()

index                          user_session  total_clicks  total_items  \
0      0  8b69208e-7c64-422d-b670-5b9d82d25606             7            2   
1      1  8068596c-a2b8-46ca-8791-e10d70520a1f             7            2   
2      2  4d153217-e06d-4581-8924-702c7f757610             2           22   
3      3  e65a6869-fb45-4f34-8482-c80278676d0f             2            2   
4      4  d8206742-06d7-42a2-8163-d70f9dd6cb18             5            1   

   total_cats  max_dwell  mean_dwell  total_duration  click_rate  day_of_week  \
0           1   0.511508    5.995878          166.71    0.072524            1   
1           1   7.626576    3.529165            0.43    0.204248            0   
2           1   3.095637  180.977516            0.94    0.045982            4   
3           1   1.089128    4.338280            0.42    0.028476            0   
4           1   0.329814    3.180891            2.49    0.094812            5   

   ...  prod_buys_freqs  is_purchase       start_time_ts         end_time_ts  \
0  ...         0.000690            0 2019-10-07 00:08:28 2019-10-26 16:08:29   
1  ...         0.000690            0 2019-10-01 15:08:40 2019-10-15 20:03:18   
2  ...         0.000690            0 2019-10-14 21:17:24 2019-10-23 21:56:30   
3  ...         0.000739            0 2019-10-05 21:54:13 2019-10-17 13:54:29   
4  ...         0.000690            0 2019-10-09 13:33:22 2019-10-29 11:51:04   

   hour  time_spec_points  total_inventory  give_big_discount  \
0     0                 0               10                  0   
1    15                15               10                  0   
2    21                26               50                  0   
3    21                26                5                  0   
4    13                13                7                  0   

   prod_views_buys_ratio   day_name  
0               1.369726     Monday  
1               1.371440    Tuesday  
2               1.369958     Monday  
3               1.279051   Saturday  
4               1.370596  Wednesday  

[5 rows x 30 columns]

In [14]:
def get_loyalty_points(day_name,total_duration):
    if (day_name=='Sunday') & (total_duration>3):
        return 10*(int(total_duration) - 3) 
    else:
        return 0
    
df['loyalty_points'] = df[['day_name','total_duration']].apply(lambda x: get_loyalty_points(*x), axis=1)
df.head()

index                          user_session  total_clicks  total_items  \
0      0  8b69208e-7c64-422d-b670-5b9d82d25606             7            2   
1      1  8068596c-a2b8-46ca-8791-e10d70520a1f             7            2   
2      2  4d153217-e06d-4581-8924-702c7f757610             2           22   
3      3  e65a6869-fb45-4f34-8482-c80278676d0f             2            2   
4      4  d8206742-06d7-42a2-8163-d70f9dd6cb18             5            1   

   total_cats  max_dwell  mean_dwell  total_duration  click_rate  day_of_week  \
0           1   0.511508    5.995878          166.71    0.072524            1   
1           1   7.626576    3.529165            0.43    0.204248            0   
2           1   3.095637  180.977516            0.94    0.045982            4   
3           1   1.089128    4.338280            0.42    0.028476            0   
4           1   0.329814    3.180891            2.49    0.094812            5   

   ...  is_purchase       start_time_ts         end_time_ts  hour  \
0  ...            0 2019-10-07 00:08:28 2019-10-26 16:08:29     0   
1  ...            0 2019-10-01 15:08:40 2019-10-15 20:03:18    15   
2  ...            0 2019-10-14 21:17:24 2019-10-23 21:56:30    21   
3  ...            0 2019-10-05 21:54:13 2019-10-17 13:54:29    21   
4  ...            0 2019-10-09 13:33:22 2019-10-29 11:51:04    13   

   time_spec_points  total_inventory  give_big_discount  \
0                 0               10                  0   
1                15               10                  0   
2                26               50                  0   
3                26                5                  0   
4                13                7                  0   

   prod_views_buys_ratio   day_name  loyalty_points  
0               1.369726     Monday               0  
1               1.371440    Tuesday               0  
2               1.369958     Monday               0  
3               1.279051   Saturday               0  
4               1.370596  Wednesday               0  

[5 rows x 31 columns]

In [15]:
df['loyalty_points'].unique()

array([   0, 2650, 3130,   10,   40,  120,   50,   20,  540, 3000, 2610,
         90,  680,  170,  240,   30,  190,  250, 2140, 3020,  580,   60,
       2750, 2290,  140, 1030,  160,  380,  110,  940,   80, 3090, 2280,
         70, 1910, 1480,  130, 3060, 3990,  550, 1170,  530, 2790,  150,
        210, 2430,  390,  300,  100, 2500,  220,  180,  290,  970, 2860,
       1540, 3330, 1450,  230, 2260,  750,  280,  410, 3720,  360, 1620,
        780, 3040,  800, 2420,  200,  270, 2960, 2520,  470,  370, 1350,
       1720,  930,  790, 2720, 2010,  830,  720,  520,  690, 1780, 2370,
       1460, 1760, 2320,  810,  620, 2560, 2440, 1890, 2590, 3290,  450,
       3200,  430, 2240, 1290,  980,  610, 1300, 2930, 3010, 2480, 2040,
       3580, 1380, 1580,  560, 4120, 3440, 1420,  440, 1140, 2620, 1370,
       2100,  330,  920, 1130,  460, 2880,  480, 1710,  320, 2740, 1800,
        600, 2810, 1000, 1490, 2000, 2080, 2940, 3550, 1950,  490, 3420,
       1940, 1430, 1180,  260, 1660, 2030,  950, 22

### **<span style="color:#483D8B;">Submission File</span>**

In [16]:
col=['total_duration','is_special_day','prod_views_freqs','prod_buys_freqs','give_big_discount',
     'prod_views_buys_ratio','loyalty_points','total_inventory','time_spec_points']
df[col].to_csv('infocept_submission.csv',index=False)